### DISCUSSION TITLE
Cannot pull calibration information from UR30 as no connection is established

### DISCUSSION THREAD

**Discussion by sachinkundu** [UniversalRobots/Universal_Robots_ROS2_Driver#1185] [Q&A]
I have Ubuntu 24.04.1 LTS running on a laptop(not a VM). ROS_DISTRO = jazzy

PolyScope version = 5.17.3

I installed jazzy version of driver using ``sudo apt-get install ros-${ROS_DISTRO}-ur``

I have connected an ethernet cable from the laptop to the control box.

I go to Hambunger > Settings > System > Network and after putting the Admin password set the network method to **Static Address** as

```
IP address: 192.168.1.102
Subnet mask: 255.255.255.0
Default gateway: 192.168.1.1
Preferred DNS server: 192.168.1.1
Alternative DNS server: 0.0.0.0
```

I then set wired connection on the laptop side to 

```
Address: 192.168.1.101
Netmask: 255.255.255.0
Gateway: 192.168.1.1
```

When I verify with `ping 192.168.1.102`it WORKS!

Even though on the Network page on the teach pendant is says **Not connected to network**

I ignored this error and proceeded with the installation by installing externalcontrol urcap as described in [Installing a URCap on a e-Series robot](https://docs.universal-robots.com/Universal_Robots_ROS2_Documentation/doc/ur_robot_driver/ur_robot_driver/doc/installation/install_urcap_e_series.html#installing-a-urcap-on-a-e-series-robot) 

The Robot program has a single statement with **ExternalControl** and pointing to Host IP: 192.168.1.101 which is the laptop ip.

Next I run 
 ``$ ros2 launch ur_calibration calibration_correction.launch.py \
robot_ip:=192.168.1.102 target_filename:="${HOME}/my_robot_calibration.yaml"``

but it fails to connect and the console stream says

```
sachin@sachin-xpose-ubuntu:~$ ros2 launch ur_calibration calibration_correction.launch.py robot_ip:=192.168.1.102 target_filename:="${HOME}/my_robot_calibration.yaml"
[INFO] [launch]: All log files can be found below /home/sachin/.ros/log/2024-11-12-09-43-57-005948-sachin-xpose-ubuntu-8680
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [calibration_correction-1]: process started with pid [8683]
[calibration_correction-1] [WARN] [1731397437.070674671] [UR_Client_Library:]: Failed to read from stream, reconnecting in 1 seconds...
[calibration_correction-1] [ERROR] [1731397574.143768702] [UR_Client_Library:]: Failed to connect to robot on IP 192.168.1.102. Please check that the robot is booted and reachable on 192.168.1.102. Retrying in 10 seconds
```

The robot is surely booted. On the laptop side I have disabled the firewall with `sudo ufw disable`

What steps do I need to do differently? I have not started the driver on the laptop side neither have I started the program from teach pendant.

I continued onward ignoring pulling the calibration file from robot with the next step ``ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur5e robot_ip:=192.168.56.101``

But that fails with 
```
...
[spawner-8] [FATAL] [1731400921.785851173] [spawner_joint_state_broadcaster]: Could not contact service /controller_manager/list_controllers
[spawner-9] [FATAL] [1731400921.820059055] [spawner_forward_position_controller]: Could not contact service /controller_manager/list_controllers
...
....
[dashboard_client-2] [ERROR] [1731401190.911485103] [UR_Client_Library:]: Failed to connect to robot on IP 192.168.1.102. Please check that the robot is booted and reachable on 192.168.1.102. Retrying in 10 seconds
```
Seems like a networking issue or are the steps to be done in another order?

**Comment by KoushikKaranGeethaNagaraj** [comment]
Hello @sachinkundu 

I am currently facing the same problem. Were you able to figure out the root cause?

**Reply by sachinkundu** [reply]
No, gave up. It was a loaner robot so have not tried it since.

**Reply by urfeex** [reply]
Sorry, I missed that discussion earlier. @KoushikKaranGeethaNagaraj could you please open a new **issue** describing your setup, especially:

* how is the PC physically connected to the robot?
* What operating system are you running?
* etc...

Our issue template should ask the relevant questions.
